lms/djangoapps/django_comment_client/utils.py

```diff
@@ -799,28 +819,113 @@ def is_commentable_divided(course_key, commentable_id):
+        # inline discussions are divided by default
         ans = True
 
     log.debug(u"is_commentable_divided(%s, %s) = {%s}", course_key, commentable_id, ans)
     return ans
 
 
+def course_discussion_division_enabled(course_discussion_settings):
+    """
+    Are discussions divided for the course represented by this instance of
+    course_discussion_settings? This method looks both at
+    course_discussion_settings.division_scheme, and information about the course
+    state itself (For example, are cohorts enabled? And are there multiple
+    enrollment tracks?).
+
+    Args:
+        course_discussion_settings: CourseDiscussionSettings model instance
+
+    Returns: True if discussion division is enabled for the course, else False
+    """
+    return _get_course_division_scheme(course_discussion_settings) != NOT_DIVIDED
+
+
+def _get_course_division_scheme(course_discussion_settings):
+    division_scheme = course_discussion_settings.division_scheme
+    if division_scheme == DIVIDED_BY_COHORT and not is_course_cohorted(course_discussion_settings.course_id):
+        division_scheme = NOT_DIVIDED
+    elif (
+        division_scheme == DIVIDED_BY_ENROLLMENT_TRACK and
+        len(_get_enrollment_track_groups(course_discussion_settings.course_id)) <= 1
+    ):
+        division_scheme = NOT_DIVIDED
+    return division_scheme
+
+
+def get_group_name(group_id, course_discussion_settings):
+    """
+    Given a specified comments_service group_id, returns the learner-facing
+    name of the Group. If no such Group exists for the specified group_id
+    (taking into account the division_scheme and course specified by course_discussion_settings),
+    returns None.
+    Args:
+        group_id: the group_id as used by the comments_service code
+        course_discussion_settings: CourseDiscussionSettings model instance
+
+    Returns: learner-facing name of the Group, or None if no such group exists
+    """
+    group_names_by_id = get_group_names_by_id(course_discussion_settings)
+    return group_names_by_id[group_id] if group_id in group_names_by_id else None
+
+
+def get_group_names_by_id(course_discussion_settings):
+    """
+    Creates of a dict of group_id to learner-facing group names, for the division_scheme
+    in use as specified by course_discussion_settings.
+    Args:
+        course_discussion_settings: CourseDiscussionSettings model instance
+
+    Returns: dict of group_id to learner-facing group names. If no division_scheme
+    is in use, returns an empty dict.
+    """
+    division_scheme = _get_course_division_scheme(course_discussion_settings)
+    course_key = course_discussion_settings.course_id
+    if division_scheme == DIVIDED_BY_COHORT:
+        return get_cohort_names(courses.get_course_by_id(course_key))
+    elif division_scheme == DIVIDED_BY_ENROLLMENT_TRACK:
+        return {-1 * group.id: group.name for group in _get_enrollment_track_groups(course_key)}
```

Comment:
Same as above, I think this could probably use the comment about the magic -1 here for future contributors